firefox_desktop/views/crash_usage.view.lkml

```diff
@@ -0,0 +1,369 @@
+    sql_end: TIMESTAMP(${submission_date}) ;;
+    intervals: [day, week, month, quarter, year]
+  }
+
+  dimension: application_name {
+    type: string
+    sql: ${TABLE}.app_name ;;
+    description: "Name of the application. Usually Firefox."
+  }
+
+  dimension: version {
+    type: string
+    sql: mozfun.norm.browser_version_info(${TABLE}.app_version).version ;;
+    description: "The full version of the application, as a string."
+    group_label: "Browser Version"
+  }
+
+  dimension: major_version {
+    type: number
+    sql: mozfun.norm.browser_version_info(${TABLE}.app_version).major_version ;;
+    description: "The major version of the application, as a number."
+    group_label: "Browser Version"
+  }
+
+  dimension: minor_version {
+    type: number
+    sql: mozfun.norm.browser_version_info(${TABLE}.app_version).minor_version ;;
+    description: "The minor version of the application, as a number."
+    group_label: "Browser Version"
+  }
+
+  dimension: patch_version {
+    type: number
+    sql: mozfun.norm.browser_version_info(${TABLE}.app_version).patch_version ;;
+    description: "The patch version of the application, as a number."
+    group_label: "Browser Version"
+  }
+
+  dimension: is_major_release {
+    type: yesno
+    sql: mozfun.norm.browser_version_info(${TABLE}.app_version).is_major_release ;;
+    description: "Whether this was a major version release (*.*.0)."
+    group_label: "Browser Version"
+  }
+
+  dimension: build_id {
+    type: string
+    sql: ${TABLE}.app_build_id ;;
+    description: "Build ID."
+  }
+
+  dimension: os_name {
+    type: string
+    sql: CASE ${TABLE}.os
+          WHEN "Darwin" THEN "Darwin"
+          WHEN "Linux" THEN "Linux"
+          WHEN "Windows_NT" THEN "Windows_NT"
+          ELSE "Other" END;;
+    description: "OS name, normalized to the top three (Windows_NT, Linux, or Darwin)."
+  }
+
+  dimension: os_version {
+    type: string
+    sql: ${TABLE}.normalized_os_version ;;
+    description: "OS version."
+  }
+
+  dimension: channel {
+    type: string
+    sql: ${TABLE}.normalized_channel ;;
+    description: "Release channel for the associated installation."
+  }
+
+  dimension: country_code {
+    type: string
+    sql: ${TABLE}.country ;;
+    hidden: yes
+  }
+
+  dimension: active_hours_sum {
+    type: number
+    sql: LEAST(${TABLE}.active_hours_sum, 24) ;;  # Filter out too-large values for active hours
+    description: "Total active hours for this client on this day, derived from active tickets (see the probe dictionary for details)."
+  }
+
+  measure: client_count {
+    type: count_distinct
+    sql: ${client_id} ;;
+    description: "Count of unique profiles."
+  }
+
+  measure: active_hours {
+    type: sum
+    sql: ${active_hours_sum} ;;
+    description: "Total active hours, derived from active ticks (see the probe dictionary for details)."
+  }
+
+  measure: main_crashes {
+    type: sum
+    sql: ${crashes_daily.main_crash_count} ;;
```

Comment:
I think it's weird that you can reference a different view from this one without an explicit include or something 🤔 Why not define and get these from the view in the above Explore join?

Reply:
Good question. Because what we actually want to calculate is `crash_rate`, which takes a value from both the `crashes_daily` table (total crashes) and `clients_daily` (active hours). `crash_rate = crashes / active_hours`. Given that, we have two options:
- Pull the `active_hours` value into the measure in `crashes_daily`
- Pull the `crash_count` value into the measure in `clients_daily`

I opted for the latter, since that let me define a single view file (and then we use the dimensions defined in `clients_daily`, rather than any of those defined in `crashes_daily`).